File: client/Classes/Page/PageEx.cpp
//
//  Chapter.cpp
//  cocos2dx
//
//  Created by Cocos2d on 12-7-18.
//  Copyright (c) 2012年 厦门雅基软件有限公司. All rights reserved.
//

#include "PageEx.h"
#include "extensions/CCBReader/CCBReader.h"
#include "extensions/CCBReader/CCNodeLoaderLibrary.h"

USING_NS_CC;

#define TAG_MENU 1000
#define TAG_NEXT 1001
#define TAG_ATTACK 1002

#define TAG_Lab_Title       11
#define TAG_Lab_Content     12
#define TAG_Sprite_Bg       21
#define TAG_Sprite_Monster  22

PageEx *PageEx::create(cocos2d::CCObject * pOwner)
{
    cocos2d::extension::CCNodeLoaderLibrary * ccNodeLoaderLibrary = cocos2d::extension::CCNodeLoaderLibrary::newDefaultCCNodeLoaderLibrary();
    
    ccNodeLoaderLibrary->registerCCNodeLoader("PageEx", PageExLoader::loader());
    
    cocos2d::extension::CCBReader * ccbReader = new cocos2d::extension::CCBReader(ccNodeLoaderLibrary);
    ccbReader->autorelease();
    
    CCNode * pNode = ccbReader->readNodeGraphFromFile("pub/", "ccb/pageEx.ccbi", pOwner);
    
    PageEx *pPage = static_cast<PageEx *>(pNode);
    return pPage;
}

bool PageEx::init()
{
    CCLayer::init();
    return true;
}

File: server/src/browser/client/Classes/Page/PageEx.js
var lc = lc = lc || {};

lc.PageEx = cc.Layer.extend({});

lc.PageExLoader = cc.LayerLoader.extend({
    _createCCNode:function (parent, ccbReader) {
        return lc.PageEx.create();
    }
});

lc.PageExLoader.loader = function () {
    return new lc.PageExLoader();
};

lc.PageEx.create = function ()
{
    var ret = new lc.PageEx();
    if (ret && ret.init()) {
        return ret;
    }
    return null;
};

lc.PageEx.createLoader = function (pOwner) {
    var ccNodeLoaderLibrary = cc.NodeLoaderLibrary.newDefaultCCNodeLoaderLibrary();

    ccNodeLoaderLibrary.registerCCNodeLoader("PageEx", lc.PageExLoader.loader());

    var ccbReader = new cc.CCBReader(ccNodeLoaderLibrary);

    var pNode = ccbReader.readNodeGraphFromFile("../Resources/","/ccb/fuzzy.ccbi");

    return pNode;
};
